pkg/oonimkall/taskrunner.go

```diff
@@ -180,47 +179,23 @@ func (r *runnerForTask) Run(rootCtx context.Context) {
 
 	builder.SetCallbacks(&runnerCallbacks{emitter: r.emitter})
 
-	// TODO(bassosimone): replace the following code with an
-	// invocation of the targetloading.Loader. Since I am making these
-	// changes before a release and I've already changed the
-	// code a lot, I'd rather avoid changing it even more,
-	// for the following reason:
-	//
-	// If we add and call targetloading.Loader here, this code will
-	// magically invoke check-in for InputOrQueryBackend,
-	// which we need to make sure the app can handle. This is
-	// the main reason why now I don't fill like properly
-	// fixing this code and use targetloading.Loader: too much work
-	// in too little time, so mistakes more likely.
-	//
-	// In fact, our current app assumes that it's its
-	// responsibility to load the inputs, not oonimkall's.
-	switch builder.InputPolicy() {
-	case model.InputOrQueryBackend, model.InputStrictlyRequired:
-		if len(r.settings.Inputs) <= 0 {
-			r.emitter.EmitFailureStartup("no input provided")
-			return
-		}
-	case model.InputOrStaticDefault:
-		if len(r.settings.Inputs) <= 0 {
-			inputs, err := targetloading.StaticBareInputForExperiment(r.settings.Name)
-			if err != nil {
-				r.emitter.EmitFailureStartup("no default static input for this experiment")
-				return
-			}
-			r.settings.Inputs = inputs
-		}
-	case model.InputOptional:
-		if len(r.settings.Inputs) <= 0 {
-			r.settings.Inputs = append(r.settings.Inputs, "")
-		}
-	default: // treat this case as engine.InputNone.
-		if len(r.settings.Inputs) > 0 {
-			r.emitter.EmitFailureStartup("experiment does not accept input")
-			return
-		}
-		r.settings.Inputs = append(r.settings.Inputs, "")
+	// Load targets. Note that, for Web Connectivity, the mobile app has
+	// already loaded inputs and provides them as r.settings.Inputs.
+	loader := builder.NewTargetLoader(&model.ExperimentTargetLoaderConfig{
+		CheckInConfig: &model.OOAPICheckInConfig{
+			// Not needed since the app already provides the
+			// inputs to use for Web Connectivity.
```

Comment:
This is not necessarily true anymore. We should probably forward the relevant fields.